allennlp/predictors/bidaf.py

```diff
@@ -38,3 +43,96 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
+                instance.add_field('answer_as_counts', field)
+
+            # When the answer is in the passage
+            elif answer_type == 'passage_span':
+                span = outputs['answer']['spans'][0]
+
+                # Convert character span indices into word span indices
+                word_span_start = None
+                word_span_end = None
+                metadata_field = MetadataField(instance['metadata'])
+                for idx, offset in enumerate(metadata_field.metadata['passage_token_offsets']):
+                    if offset[0] == span[0]:
+                        word_span_start = idx
+                    if offset[1] == span[1]:
+                        word_span_end = idx
+
+                passage = SequenceField(instance['passage'])
+                field = ListField([SpanField(word_span_start, word_span_end, passage)])
+                instance.add_field('answer_as_passage_spans', field)
+
+            # When the answer is an arithmetic calculation
+            elif answer_type == 'arithmetic':
+                # The different numbers in the passage that the model encounters
+                sequence_labels = outputs['answer']['numbers']
+
+                # TODO (Eric-Wallace) outputs['answer']['numbers'] does not
+                # include padding (like metadata), is that ok?
+                numbers_as_tokens = \
+                [Token(str(number['value'])) for number in outputs['answer']['numbers']]
+                # hack copied from https://github.com/
+                # allenai/allennlp/blob/master/allennlp/
+                # data/dataset_readers/reading_comprehension/drop.py#L232
+                numbers_as_tokens.append(Token('0'))
+                numbers_in_passage_field = \
+                TextField(numbers_as_tokens, self._dataset_reader._token_indexers) # type: ignore
+
+                # Based on ``find_valid_add_sub_expressions``, it seems
+                # like negative signs are 2 instead of -1.
+                # The numbers in the passage are given signs, that's what
+                # we are labeling here
+                labels = []
+                for label in sequence_labels:
+                    if label['sign'] == -1:
+                        labels.append(2)
+                    else:
+                        labels.append(label['sign'])
+                labels.append(0) # 0 stands for "not included"
+
+                field = ListField([SequenceLabelField(labels, numbers_in_passage_field)])
+                instance.add_field('answer_as_add_sub_expressions', field)
+
+            # When the answer is in the question
+            elif answer_type == 'question_span':
+                span = outputs['answer']['spans'][0]
+
+                # Convert character span indices into word span indices
+                word_span_start = None
+                word_span_end = None
+                for idx, offset in enumerate(MetadataField(instance['metadata']). \
```

Comment:
same issue as before

Reply:
ill just add an ignore for most of these. I tried wrapping things in MetadataField to make mypy stop, but that didn't seem to do anything.